internal/cortex/chunk/cache/memcached_client.go

```diff
@@ -107,9 +111,19 @@ func NewMemcachedClient(cfg MemcachedClientConfig, name string, r prometheus.Reg
 	client.Timeout = cfg.Timeout
 	client.MaxIdleConns = cfg.MaxIdleConns
 
-	dnsProviderRegisterer := prometheus.WrapRegistererWithPrefix("cortex_", prometheus.WrapRegistererWith(prometheus.Labels{
-		"name": name,
-	}, r))
+	var addressProvider clientconfig.AddressProvider
+	if cfg.AutoDiscovery {
+		addressProvider = memcacheDiscovery.NewProvider(
+			logger,
+			extprom.WrapRegistererWithPrefix("thanos_memcached_", r),
```

Comment:
Is it better to use the same metric prefix here? `cortex_`

Reply:
It was copied from thanos store code. I did as you suggested.